File: files/2025-03-17/CSEC 16032025.R
library(tidyverse)
library(readxl)

path = "files/CHALLENGE 1205.xlsx"
input = read_excel(path, range = "B3:B13")
test  = read_excel(path, range = "D3:G7")

result = input %>%
  separate(`SUB-DEPARTMENT NAMES`, into = c("sub_department", "department"), sep = "-", extra = "merge", remove = F) %>%
  mutate(rn = row_number(), .by = sub_department) %>%
  select(-department) %>%
  pivot_wider(names_from = sub_department, values_from = `SUB-DEPARTMENT NAMES`) %>%
  select(-rn)

all.equal(result, test, check.attributes = FALSE)
#> [1] TRUE

File: files/2025-03-17/CSEC 16032025.py
import pandas as pd

path = "CHALLENGE 1205.xlsx"
input_data = pd.read_excel(path, usecols="B", skiprows=2, nrows=11)
test = pd.read_excel(path, usecols="D:G", skiprows=2, nrows=5)

result = (
    input_data["SUB-DEPARTMENT NAMES"]
    .str.split("-", n=1, expand=True)
    .rename(columns={0: "sub_department", 1: "department"})
)
result["SUB-DEPARTMENT NAMES"] = input_data["SUB-DEPARTMENT NAMES"]
result["rn"] = result.groupby("sub_department").cumcount() + 1
result = result.drop(columns=["department"]).pivot(index="rn", columns="sub_department", values="SUB-DEPARTMENT NAMES").reset_index(drop=True)

print(result.equals(test))
